### PR TITLE
configuration bot

### PR DESCRIPTION
**Pull request by monoskribt**

### CODE REVIEW

File: src/main/java/com/spotifytelegramdeliverymessage/controller/BotController.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        }
+
+        String message = update.getMessage().getText();
+        String id = update.getMessage().getChatId().toString();
+        String username = update.getMessage().getChat().getUserName();
+
+
+
+        if (message.startsWith(START)) {
+            try {
```
Comment (syevtushok) [inline src/main/java/com/spotifytelegramdeliverymessage/controller/BotController.java]: use one try-catch

File: src/main/java/com/spotifytelegramdeliverymessage/controller/BotController.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+                botService.sendWelcomeMessage(id, username);
+            } catch (TelegramApiException e) {
+                throw new RuntimeException(e);
```
Comment (syevtushok) [inline src/main/java/com/spotifytelegramdeliverymessage/controller/BotController.java]: don't throw RuntimeException

File: src/main/java/com/spotifytelegramdeliverymessage/service/impl/BotServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    @Autowired
+    private AbsSender absSender;
+
+    @Autowired
+    private UserService userService;
+
+    @Autowired
+    private EmailService emailService;
+
+    private final Map<String, Integer> confirmationCodeMap = new HashMap<>();
```
Comment (syevtushok) [inline src/main/java/com/spotifytelegramdeliverymessage/service/impl/BotServiceImpl.java]: Maybe store it in a table

File: src/main/java/com/spotifytelegramdeliverymessage/service/impl/EmailServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+        message.setTo(toEmail);
+        message.setSubject("Confirmation Code");
+        message.setText("Your confirmation code is: " + confirmationCode);
+
+        javaMailSender.send(message);
+    }
+
+    @Override
+    public int generateConfirmationCode() {
+        int lowerLimit = 100000;
```
Comment (syevtushok) [inline src/main/java/com/spotifytelegramdeliverymessage/service/impl/EmailServiceImpl.java]: move it to const

File: src/main/resources/application.properties [hunk 5]
```diff
@@ -0,0 +1,18 @@
+spring.profiles.active=dev
+spring.application.name=spotify-telegram-delivery-message
+
+## Set up bot
+bot.name=spotify_message_bot
+bot.token=7595434379:AAEXW-hG3SBUOArUGjJQNWhWynZrQwJ-EKM
```
Comment (syevtushok) [inline src/main/resources/application.properties]: don't commit it

File: src/test/java/com/spotifytelegramdeliverymessage/SpotifyTelegramDeliveryMessageApplicationTests.java [hunk 6]
```diff
@@ -0,0 +1,13 @@
+package com.spotifytelegramdeliverymessage;
+
+import org.junit.jupiter.api.Test;
+import org.springframework.boot.test.context.SpringBootTest;
+
+@SpringBootTest
```
Comment (syevtushok) [inline src/test/java/com/spotifytelegramdeliverymessage/SpotifyTelegramDeliveryMessageApplicationTests.java]: write a full test or remove it completely

File: src/main/java/com/spotifytelegramdeliverymessage/controller/BotController.java (hunk 1 above)
Comment (monoskribt) [reply src/main/java/com/spotifytelegramdeliverymessage/controller/BotController.java]: Done

File: src/main/java/com/spotifytelegramdeliverymessage/controller/BotController.java (hunk 2 above)
Comment (monoskribt) [reply src/main/java/com/spotifytelegramdeliverymessage/controller/BotController.java]: Replaced by Exception

File: src/main/java/com/spotifytelegramdeliverymessage/service/impl/BotServiceImpl.java (hunk 3 above)
Comment (monoskribt) [reply src/main/java/com/spotifytelegramdeliverymessage/service/impl/BotServiceImpl.java]: Created new Column on DB with name "code"

File: src/main/java/com/spotifytelegramdeliverymessage/service/impl/EmailServiceImpl.java (hunk 4 above)
Comment (monoskribt) [reply src/main/java/com/spotifytelegramdeliverymessage/service/impl/EmailServiceImpl.java]: Done

File: src/main/resources/application.properties (hunk 5 above)
Comment (monoskribt) [reply src/main/resources/application.properties]: moved it to another file

File: src/test/java/com/spotifytelegramdeliverymessage/SpotifyTelegramDeliveryMessageApplicationTests.java (hunk 6 above)
Comment (monoskribt) [reply src/test/java/com/spotifytelegramdeliverymessage/SpotifyTelegramDeliveryMessageApplicationTests.java]: deleted so far

File: src/main/java/com/spotifytelegramdeliverymessage/controller/BotController.java [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+        try {
+            if (message.startsWith(START)) {
+                botService.sendWelcomeMessage(id, username);
+            } else if (message.startsWith(SUBSCRIBE)) {
+                botService.subscribe(id, username, message);
+            } else if (message.startsWith(CONFIRM)) {
+                botService.confirmation(id, username, message);
+            }
+        } catch (TelegramApiException e) {
+            System.err.println("Something is wrong: " + e.getMessage());
```
Comment (syevtushok) [inline src/main/java/com/spotifytelegramdeliverymessage/controller/BotController.java]: not the best way but it's okay for now

File: src/main/java/com/spotifytelegramdeliverymessage/service/impl/BotServiceImpl.java [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+    @Autowired
+    private AbsSender absSender;
+
+    @Autowired
+    private UserService userService;
+
+    @Autowired
+    private EmailService emailService;
+
+    private final Map<String, String> emailByChatId = new HashMap<>();
```
Comment (syevtushok) [inline src/main/java/com/spotifytelegramdeliverymessage/service/impl/BotServiceImpl.java]: what if your app is restarted? Consider adding this info also to db

File: src/main/java/com/spotifytelegramdeliverymessage/controller/BotController.java (hunk 7 above)
Comment (monoskribt) [reply src/main/java/com/spotifytelegramdeliverymessage/controller/BotController.java]: I have idea to create later Logger class and save logs to file

File: src/main/java/com/spotifytelegramdeliverymessage/service/impl/BotServiceImpl.java (hunk 8 above)
Comment (monoskribt) [reply src/main/java/com/spotifytelegramdeliverymessage/service/impl/BotServiceImpl.java]: Done

**Review by syevtushok — APPROVED** [review]

**Pull request merged**